internal/storage/storage_handle_test.go

```diff
@@ -253,6 +254,186 @@ func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_WithGRPCClientPro
+	}{
+		{
+			name:                 "ReadStallRetryEnabled",
+			enableReadStallRetry: true,
+		},
+		{
+			name:                 "ReadStallRetryDisabled",
+			enableReadStallRetry: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		testSuite.Run(tc.name, func() {
+			sc := storageutil.GetDefaultStorageClientConfig()
+			sc.ReadStallRetryConfig.Enable = tc.enableReadStallRetry
+
+			storageClient, err := createHTTPClientHandle(context.Background(), &sc)
+
+			assert.Nil(testSuite.T(), err)
+			assert.NotNil(testSuite.T(), storageClient)
+		})
+	}
+}
+
+func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_ReadStallInitialReqTimeout() {
+	testCases := []struct {
+		name              string
+		initialReqTimeout time.Duration
+	}{
+		{
+			name:              "ShortTimeout",
+			initialReqTimeout: 1 * time.Millisecond,
+		},
+		{
+			name:              "LongTimeout",
+			initialReqTimeout: 10 * time.Second,
+		},
+	}
+
+	for _, tc := range testCases {
+		testSuite.Run(tc.name, func() {
+			sc := storageutil.GetDefaultStorageClientConfig()
+			sc.ReadStallRetryConfig.Enable = true
+			sc.ReadStallRetryConfig.InitialReqTimeout = tc.initialReqTimeout
+
+			storageClient, err := createHTTPClientHandle(context.Background(), &sc)
+
+			assert.Nil(testSuite.T(), err)
+			assert.NotNil(testSuite.T(), storageClient)
+		})
+	}
+}
+
+func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_ReadStallMinReqTimeout() {
+	testCases := []struct {
+		name          string
+		minReqTimeout time.Duration
+	}{
+		{
+			name:          "ShortTimeout",
+			minReqTimeout: 1 * time.Millisecond,
+		},
+		{
+			name:          "LongTimeout",
+			minReqTimeout: 10 * time.Second,
+		},
+	}
+
+	for _, tc := range testCases {
+		testSuite.Run(tc.name, func() {
+			sc := storageutil.GetDefaultStorageClientConfig()
+			sc.ReadStallRetryConfig.Enable = true
+			sc.ReadStallRetryConfig.MinReqTimeout = tc.minReqTimeout
+
+			storageClient, err := createHTTPClientHandle(context.Background(), &sc)
+
+			assert.Nil(testSuite.T(), err)
+			assert.NotNil(testSuite.T(), storageClient)
+		})
+	}
+}
+
+func (testSuite *StorageHandleTest) TestCreateHTTPClientHandle_ReadStallReqIncreaseRate() {
+	testCases := []struct {
+		name            string
+		reqIncreaseRate float64
+		expectErr       bool
+	}{
+		{
+			name:            "NegativeRate",
+			reqIncreaseRate: -0.5,
+			expectErr:       true,
+		},
+		{
+			name:            "ZeroRate",
+			reqIncreaseRate: 0.0,
+			expectErr:       true,
+		},
+		{
+			name:            "PositiveRate",
+			reqIncreaseRate: 1.5,
+			expectErr:       false,
+		},
+	}
+
+	for _, tc := range testCases {
+		testSuite.Run(tc.name, func() {
+			sc := storageutil.GetDefaultStorageClientConfig()
+			sc.ReadStallRetryConfig.Enable = true
+			sc.ReadStallRetryConfig.ReqIncreaseRate = tc.reqIncreaseRate
+
+			storageClient, err := createHTTPClientHandle(context.Background(), &sc)
+
+			if tc.expectErr {
+				assert.NotNil(testSuite.T(), err)
```

Comment:
Do we know what the expected error type or string is? If yes, we should use that for more accurate checking,.